docs/developer/file-structure.rst

```diff
@@ -28,58 +29,184 @@ The PyMedPhys repository is generally structured as follows:
 
 .. code-block:: bash
 
    pymedphys/
+   |
    |-- src/pymedphys
    |   |
    |   |-- dicom/
    |   |   |-- __init__.py
    |   |   |
    |   |   |-- _level1/
    |   |   |   |-- __init__.py
-   |   |   |   |-- a_python_source_code_file.py
-   |   |   |   |-- another_python_source_code_file.py
+   |   |   |   |-- d1a.py
+   |   |   |   |-- d1b.py
    |   |   |
    |   |   |-- _level2/
+   |   |   |   |-- __init__.py
+   |   |   |   |-- d2a.py
+   |   |   |   |-- d2b.py
    |   |   |
    |   |   |-- _level3/
+   |   |   |   |-- __init__.py
+   |   |   |   |-- d3a.py
    |   |   |
    |   |   |-- _level4/
+   |   |   |   |-- __init__.py
+   |   |   |   |-- d4a.py
    |   |
    |   |-- gamma/
    |   |   |-- __init__.py
    |   |   |
    |   |   |-- _level1/
    |   |   |   |-- __init__.py
-   |   |   |   |-- and_yet_even_more_python_code.py
+   |   |   |   |-- g1a.py
    |   |   |
    |   |   |-- _level2/
+   |   |   |   |-- __init__.py
+   |   |   |   |-- g2a.py
+   |   |   |   |-- g2b.py
+   |   |   |   |-- g2c.py
    |   |   |
    |   |   |-- _level3/
+   |   |   |   |-- __init__.py
+   |   |   |   |-- g3a.py
    |   |   |
    |   |   |-- _level4/
+   |   |   |   |-- __init__.py
+   |   |   |   |-- g4a.py
    |   |
    |   |-- ...
+   |   
+   |-- ...
+
+For the most part, the many ``__init__.py`` files just tell Python to treat
+directories containing the files as *packages*. They form part of how
+PyMedPhys' code is brought together as an installable package or library whose
+modules can be imported.
+
+Python files within the source code should have descriptive names indicating
+the functions of the code within them. For example, ``dose.py`` in level 1 of
+``dicom`` is so-named because it contains code that interacts with DICOM RT
+Dose files. However, in order to illustrate how levelling works in PyMedPhys,
+the files in the above diagram have been named according to their level and
+module like so:
+
+``<first-letter-of-module><level number><letter-to-differentiate-files-in-the-same-module-and-level>``
+
+E.g. ``g2a.py`` is the first file in level 2 of the ``gamma`` module in the
+above diagram.
+
+The key to levelling is this: **The code contained in files of a particular
+level should only depend on code in files of lower-numbered levels. Code should
+never depend on code within files of the same level, nor of higher-numbered
+levels.**
+
+For example, ``g2a.py`` is in level 2, so code in ``g2a.py`` can depend on code
+in ``g1a.py``, because ``g1a.py`` is in level 1 (a lower-numbered level). In
+contrast code in ``g2a.py`` *cannot* depend on code in ``g2b.py`` (which is in
+the same level), ``g3a.py`` or ``g4a.py`` (which are in higher-numbered
+levels).
+
+*This philosophy applies for modules themselves as well.* It's not as explictly
+clear from the module name which level it belongs in, but there exists a file
+``setup.cfg`` which details which level each module belongs to. Once at module
```

Comment:
This is the file that is used for the new layer-linter package (import-linter) will submit that now in another branch.